pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -275,12 +289,21 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
           }
           _tableLowestLevelMaxValidBucketEndTimeMs.put(tableNameWithType, lowestLevelMaxValidBucketEndTimeMs);
         }
-        // Create delay metrics even if there's no task scheduled, this helps the case that the controller is restarted
+        // Create metrics even if there's no task scheduled, this helps the case that the controller is restarted
         // but the metrics are not available until the controller schedules a valid task
-        createOrUpdateDelayMetrics(tableNameWithType, mergeLevel, null, watermarkMs, bufferMs, bucketMs);
-        if (!isValidBucketEndTime(bucketEndMs, bufferMs, lowerMergeLevel, mergeRollupTaskMetadata)) {
-          LOGGER.info("Bucket with start: {} and end: {} (table : {}, mergeLevel : {}) cannot be merged yet",
-              bucketStartMs, bucketEndMs, tableNameWithType, mergeLevel);
+        List<String> sortedMergeLevels =
+            sortedMergeLevelConfigs.stream().map(e -> e.getKey()).collect(Collectors.toList());
+        if (processAll) {
```

Comment:
Why don't we emit both metrics in both cases? 

1. numBuckets to process is anyway going to be useful for original mode.
2. watermark may not mean much for `processAll` mode but we still can alert if there's no tasks get scheduled for new time buckets for a long time. This allows us to be able to alert on "no activity for a long time" on a single metric.

Reply:
1. For the original mode, numBuckets will be the same as the delayInNumBuckets.
2. For `processAll` mode, we can just use numBuckets to alert on "no activity for a long time"? delayInNumBuckets/watermark won't reflect the correct backlog if there are some buckets processed but don't change the watermark.

@snleee wdyt?

Reply:
cool. what will be the alert mechanism for the new metric? Probably, we alert if numBucketsToProcess > 0 for more than x hours/days?

Reply:
Yes, that correct.